src/azure-cli-core/azure/cli/core/extension.py

```diff
@@ -0,0 +1,196 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import traceback
+import json
+
+from wheel.install import WHEEL_INFO_RE
+
+import azure.cli.core.azlogging as azlogging
+
+from azure.cli.core._config import GLOBAL_CONFIG_DIR
+
+
+_CUSTOM_EXT_DIR = os.environ.get('AZURE_EXTENSION_DIR')
+EXTENSIONS_DIR = os.path.expanduser(_CUSTOM_EXT_DIR) if _CUSTOM_EXT_DIR \
+                    else os.path.join(GLOBAL_CONFIG_DIR, 'cliextensions')
+EXTENSIONS_MOD_PREFIX = 'azext_'
+
+WHL_METADATA_FILENAME = 'metadata.json'
+AZEXT_METADATA_FILENAME = 'azext_metadata.json'
+
+EXT_METADATA_MINCLIVERSION = 'azext.minCliVersion'
+EXT_METADATA_MAXCLIVERSION = 'azext.maxCliVersion'
+
+logger = azlogging.get_az_logger(__name__)
+
+logger.debug("Extension directory: '%s'", EXTENSIONS_DIR)
+
+
+class ExtensionNotInstalledException(Exception):
+    def __init__(self, extension_name):
+        super(ExtensionNotInstalledException, self).__init__(extension_name)
+        self.extension_name = extension_name
+
+    def __str__(self):
+        return "The extension {} is not installed.".format(self.extension_name)
+
+
+class Extension(object):
+
```

Comment:
In the future, I expect we'll have more extension types.
So the `WheelExtension` class extends from this base class.

Does this harm anything by keeping it in?